### DISCUSSION TITLE
Keycloak -18.0.1 issue with authorization and multiple polices/permissions

### DISCUSSION THREAD

**Discussion by GaganNoor-tieto** [keycloak/keycloak#14853] [Q&A]
Hi, we are getting issue with multiple policies/permission.

Policy A - user A with scope A,B,C works fine.
Policy B - user B with scope B,C,D gives 403.
Policy C - user C with scope C,D,E gives 403.

If User B is added to Policy A it works fine and retunes combined result of scope A,B,C,D.

For User C we need to this add this user to both policy A,B and it works fine and retunes combined result of scope A,B,C,D,E.

Both Policies are Positive and Both Permissions are Affirmative.

We want this to work independently. For user B and C to work by adding them to policy B for user B and policy C for user C.

In Evaluate page it shows as expected but from Postman it gives error

**Comment by AndriusKarpavicius** [comment]
By scope you mean scopes you assign in resource definition, or a client scope policy?

If scope you assign in resource definition, then the model  we used is:
1. Specify all scopes in resource . (e.g. Get, post, put, delete)
2. Create role based policies
3.  Create scope-based permissions for whatever scope combination you need and chose a role policy to apply. We had two permissions per resource - to distinguish between read and write. Read - get and write - post, put, delete.

**Reply by GaganNoor-tieto** [reply]
Hi Andrius, 

I have attached few of the details to help understand what we are trying to achieve.  Please let me know if any more details are required.


[KeycloakIssue.docx](https://github.com/keycloak/keycloak/files/10236370/KeycloakIssue.docx)
